sbpy/data/ephem.py

```diff
@@ -127,35 +129,137 @@ def from_horizons(cls, targetids, id_type='smallbody',
         return cls.from_table(all_eph)
 
     @classmethod
-    def from_mpc(cls, targetid, epoch, observatory='500'):
-        """
-        Load ephemerides from the
+    def from_mpc(cls, targetid, epochs=None, start=None, step=None,
+                 stop=None, number=None, location='500', **kwargs):
+        """Load ephemerides from the
         `Minor Planet Center <http://minorplanetcenter.net>`_.
 
         Parameters
         ----------
-        targetid : str, mandatory
-            target identifier
-        epochs : astropy Time instance or iterable, optional, default ``None``
-            epoch of elements; if ``None`` is provided, current date is used
-        observatory : str, optional, default ``'500'`` (geocentric)
-            location of observer
+        targetid : str
+            Target identifier.  Must be resolvable by the Minor Planet
+            Ephemeris Service [MPES]_.
+
+        epochs : string, `~astropy.time.Time`, or iterable, optional
+            Request ephemerides at these specific epochs.  If neither
+            of ``epochs`` nor ``start`` is provided, the current time
+            will be used.
+
+        start : string or `~astropy.time.Time`, optional
+            Request ephemerides starting at this epoch, superceding
+            the ``epochs`` parameter.  Strings are parsed by
+            `~astropy.time.Time` assuming a UTC scale.
+
+        step : string or `~astropy.units.Quanitity`, optional
+            Specifies the ephemeris step size when ``start`` is
+            provided.  The MPES requires integer steps with units of
+            seconds, minutes, hours, or days.  Strings are parsed by
+            `~astropy.units.Quantity`.
+
+        stop : string or `~astropy.units.Time`, optional
+            Specifies the ephemeris end time when ``start`` is
+            provided.  Supercedes ``number``, requires ``step``.
+            Strings are parsed by `~astropy.time.Time` assuming a UTC
+            scale.  The stop epoch may not be included depending on
+            the step size and floating point arithmetic.
+
+        number : int, optional
+            The number of ephemeris dates to compute.  Superceded by
+            ``stop``.
+
+        location : str, array-like, or `~astropy.coordinates.EarthLocation`, optional
```

Comment:
this is cool, I didn't know about `astropy.coordinates.EarthLocation`; will implement this too for `jplhorizons`, where currently a dictionary of coordinates is allowed

Reply:
After this is merged, we should work on getting all the overlaps into a shared code base, which will make extending `Ephem` a bit easier.